scipy/interpolate/_interpolate.py

```diff
@@ -2534,6 +2534,9 @@ def __call__(self, xi, method=None):
         xi_shape = xi.shape
         xi = xi.reshape(-1, xi_shape[-1])
 
+        # find nans in input
+        nans = [np.isnan(v).any() for v in xi]
```

Comment:
Just a thought, in case the `axis` argument of `np.any` didn't come to mind.
```suggestion
        nans = np.any(np.isnan(xi), axis=tuple(range(1, xi.ndim)))
```
Then again, `tuple(range(...))` is a bit clumsy, so I can see why you still might prefer the comprehension.

I can't tell from the code whether `xi` is always 2D? Sometimes (e.g.`xi.T`) it looks like it, elsewhere (e.g. `xi.reshape(-1, xi_shape[-1])` not so much...)  In that case, this could be simplified to:

```suggestion
        nans = np.any(np.isnan(xi), axis=-1)
```

If it is 1D or 2D, seems like an `np.atleast_2d` at the beginning could simplify things.

Reply:
I *think* `xi` is 2D at this point after the weird-looking reshape above. So your `(..., axis=-1)`suggestion should work, and if it makes the intent clearer to you, great, let's do it like this.

Reply:
I think they are equally clear. Not a strong preference; I just tend toward vectorization over explicit loops unless there's a reason not to.